packages/sources/onchain-gas/src/adapter.ts

```diff
@@ -28,9 +28,14 @@ export const makeWSHandler = (config?: Config): MakeWSHandler => {
       connection: {
         url: defaultConfig.api.baseURL,
       },
-      toSaveFromFirstMessage: (message: any) => ({
-        subscriptionId: message.params.subscription,
-      }),
+      toSaveFromFirstMessage: (message: any) => {
+        if (message.method !== 'eth_subscription' || !message.params) {
```

Comment:
```suggestion
        if (message.method !== 'eth_subscription' || !message.params) return null
```
Do you prefer brackets and such?
Null is nice as a return because it denotes we chose to return it.

Reply:
I don't have a preference on whether or not to use brackets.  I think I'm just more used to writing code with it.